be/src/storage/tablet_updates.cpp

```diff
@@ -3368,13 +3368,13 @@ Status TabletUpdates::reorder_from(const std::shared_ptr<Tablet>& base_tablet, i
         ChunkPtr base_chunk = ChunkHelper::new_chunk(base_schema, config::vector_chunk_size);
 
         Schema new_schema = ChunkHelper::convert_schema(_tablet.tablet_schema());
-        ChunkPtr new_chunk = ChunkHelper::new_chunk(new_schema, config::vector_chunk_size);
 
         for (auto& seg_iterator : seg_iterators) {
             if (seg_iterator.get() == nullptr) {
                 continue;
             }
             while (true) {
+                ChunkPtr new_chunk = ChunkHelper::new_chunk(new_schema, config::vector_chunk_size);
```

Comment:
Why to new a chunk in every time? Just set number to zero is enough.

Reply:
> Why to new a chunk in every time? Just set number to zero is enough.

Because we need to sort the chunks again, so we need to keep these chunks. If we reset it, the chunk data are change.